R/dot-net-wrapper.R

```diff
@@ -6,113 +6,47 @@
-      private$.ref <- ref
-    }
-  ),
-  active = list(
-    #' @field ref Underlying `.NET` reference (read-only)
-    ref = function(value) {
-      private$readOnlyProperty("ref", value, private$.ref)
-    }
-  ),
   private = list(
-    .ref = NULL,
-
-    # Simple way to wrap a get;set; .NET property
-    wrapProperty = function(propertyName, value, shouldSetNull = TRUE) {
-      if (missing(value)) {
-        rClr::clrGet(self$ref, propertyName)
-      } else {
-        # Problem converting reference object to `NULL`
-        if (is.null(value) && !shouldSetNull) {
-          return()
-        }
-
-        if (isOfType(type = "character", object = value)) {
-          # isOfType returns TRUE for empty `object` and enc2utf8(value) fails
-          if (length(value) > 0) {
-            value <- enc2utf8(value)
-          }
-          rClr::clrSet(self$ref, propertyName, value)
-        } else {
-          rClr::clrSet(self$ref, propertyName, value)
-        }
-      }
-    },
-
-    # Simple way to wrap a get; .NET Read-Only property
-    wrapReadOnlyProperty = function(propertyName, value) {
+    .wrapExtensionMethod = function(typename, methodName, propertyName, value) {
       if (missing(value)) {
-        rClr::clrGet(self$ref, propertyName)
+        rSharp::callStatic(typename, methodName, self)
       } else {
-        private$throwPropertyIsReadonly(propertyName)
+        private$.throwPropertyIsReadonly(propertyName)
       }
     },
-    wrapExtensionMethod = function(typename, methodName, propertyName, value) {
-      if (missing(value)) {
-        rClr::clrCallStatic(typename, methodName, self$ref)
-      } else {
-        private$throwPropertyIsReadonly(propertyName)
-      }
-    },
-    wrapExtensionMethodCached = function(typename, methodName, propertyName, cachedValue, value) {
+    .wrapExtensionMethodCached = function(typename, methodName, propertyName, cachedValue, value) {
       if (missing(value)) {
         if (is.null(cachedValue)) {
-          return(rClr::clrCallStatic(typename, methodName, self$ref))
+          return(rSharp::callStatic(typename, methodName, self))
         }
         return(cachedValue)
       } else {
-        private$throwPropertyIsReadonly(propertyName)
+        private$.throwPropertyIsReadonly(propertyName)
       }
     },
-    readOnlyProperty = function(propertyName, value, returnValue) {
+    .readOnlyProperty = function(propertyName, value, returnValue) {
       if (missing(value)) {
         returnValue
       } else {
-        private$throwPropertyIsReadonly(propertyName)
-      }
-    },
-    wrapIntegerProperty = function(propertyName, value) {
-      # Special method needed because of double to int conversion issues between R and `.NET`
-      if (missing(value)) {
-        rClr::clrGet(self$ref, propertyName)
-      } else {
-        rClr::clrSet(self$ref, propertyName, as.integer(value))
-      }
-    },
-    wrapNullableIntegerProperty = function(propertyName, value) {
```

Comment:
Instead of a separate method, `wrapProperty` has an argument `shouldSetNull`:

https://github.com/Open-Systems-Pharmacology/rSharp/blob/2204ca57c5e8e84c733f3301f4ee9407a76c768f/R/net-object.R#L44